action/protocol/staking/protocol.go

```diff
@@ -109,3 +121,33 @@ func (p *Protocol) Register(r *protocol.Registry) error {
 func (p *Protocol) ForceRegister(r *protocol.Registry) error {
 	return r.ForceRegister(protocolID, p)
 }
+
+// CreateGenesisStates creates genesis states
+func (p *Protocol) CreateGenesisStates(ctx context.Context, sm protocol.StateManager) error {
+	if p.inMemCandidates.Size() > 0 {
+		return nil
+	}
+
+	// read all candidates from stateDB
+	stateDB := sm.GetDB()
+	candPkHash, candValue, err := stateDB.Filter(factory.CandidateNameSpace, func(k, v []byte) bool {
+		return bytes.HasPrefix(k, []byte{candidate})
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(candValue) == 0 {
+		return nil
+	}
```

Comment:
can be removed. len == 0 will just pass below for loop

Reply:
updated in latest push